presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -525,30 +524,32 @@ public static LocalTime fromPrestoTime(long value)
             case Types.NUMERIC:
             case Types.DECIMAL:
                 int decimalDigits = type.getRequiredDecimalDigits();
-                int precision = columnSize + max(-decimalDigits, 0); // Map decimal(p, -s) (negative scale) to decimal(p+s, 0).
+                int precision = type.getRequiredColumnSize() + max(-decimalDigits, 0); // Map decimal(p, -s) (negative scale) to decimal(p+s, 0).
                 if (precision > Decimals.MAX_PRECISION) {
                     return Optional.empty();
                 }
                 return Optional.of(decimalColumnMapping(createDecimalType(precision, max(decimalDigits, 0))));
 
             case Types.CHAR:
             case Types.NCHAR:
-                if (columnSize > CharType.MAX_LENGTH) {
-                    if (columnSize > VarcharType.MAX_LENGTH) {
+                int charLength = type.getRequiredColumnSize();
+                if (charLength > CharType.MAX_LENGTH) {
+                    if (charLength > VarcharType.MAX_LENGTH) {
                         return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
                     }
-                    return Optional.of(varcharColumnMapping(createVarcharType(columnSize)));
+                    return Optional.of(varcharColumnMapping(createVarcharType(charLength)));
                 }
-                return Optional.of(charColumnMapping(createCharType(columnSize)));
+                return Optional.of(charColumnMapping(createCharType(charLength)));
 
             case Types.VARCHAR:
             case Types.NVARCHAR:
             case Types.LONGVARCHAR:
             case Types.LONGNVARCHAR:
-                if (columnSize > VarcharType.MAX_LENGTH) {
+                int varcharLength = type.getRequiredColumnSize();
+                if (varcharLength > VarcharType.MAX_LENGTH) {
```

Comment:
The varchar mapping could use the new `Optional`-bearing getter:

```java
return type.getColumnSize()
    .filter(length -> length <= VarcharType.MAX_LENGTH)
    .map(length -> varcharColumnMapping(createVarcharType(length)))
    .or(() -> Optional.of(varcharColumnMapping(createUnboundedVarcharType())));
```

Reply:
maybe it is just me, but I find it hard to read code using `Optional#filter`. That's too clever for me.
(i have no problem _writing_ such code though)

Reply:
For one this simple, I think it's readable, and the pattern is easy to learn, but it's a matter of taste. (And for anything more complicated (like the mapping for `char`), it's a definitely bad idea.)

I still prefer using `Optional` chaining, but this might be a more readable way to do that:

```java
return type.getColumnSize()
    .map(length ->
        length <= VarcharType.MAX_LENGTH
            ? createVarcharType(length)
            : createUnboundedVarcharType())
    .map(StandardColumnMappings::varcharColumnMapping)
```

(Though neither this nor the current solution works if any JDBC supports unbounded varchar and reports them as having null column size. I assume we don't know any such JDBC.)